pybamm/experiments/experiment.py

```diff
@@ -221,23 +251,47 @@ def read_string(self, cond, drive_cycles):
                 cond_list = cond.split()
                 idx_for = cond_list.index("for")
                 idx_until = cond_list.index("or")
+
                 electric = self.convert_electric(cond_list[:idx_for])
+
                 time = self.convert_time_to_seconds(cond_list[idx_for + 1 : idx_until])
-                events = self.convert_electric(cond_list[idx_until + 2 :])
+
+                # remove temperture part of string
+                reduced_cond_list = cond_list[idx_until + 2 :]
+                if "at" in reduced_cond_list:
+                    at_idx = reduced_cond_list.index("at")
+                    reduced_cond_list = reduced_cond_list[:at_idx]
+
+                events = self.convert_electric(reduced_cond_list)
```

Comment:
this block of code looks like it's repeated 3 times, is this really the best way of doing this? If so, could it be a function?

I think you should be able to regex match `"at XoC"` where `X` is any number, record temperature, and remove that from the string for all cases

Reply:
i.e. find it and then do `.replace("at 25oC", "")`